File: Algorithms/Go/selection_sort.go
package main

import "fmt"

func selectionSort(arr []int) {
	for i := 0; i < len(arr)-1; i++ {
		minIndex := i

		for j := i + 1; j < len(arr); j++ {
			if arr[j] < arr[minIndex] {
				minIndex = j
			}
		}

		arr[minIndex], arr[i] = arr[i], arr[minIndex]
	}
}

func main() {
	arr := []int{5, 4, 3, 2, 1}
	selectionSort(arr)
	fmt.Println(arr)
}

File: Algorithms/Python/selection_sort.py
from __future__ import annotations

from typing import TypeVar


T = TypeVar("T", int, float)


def selection_sort(arr: list[int]) -> None:
    for i, num in enumerate(arr):
        minimum = arr[i]
        index = i
        for j in range(i, len(arr)):
            if minimum > arr[j]:
                index = j
                minimum = arr[j]
        arr[i], arr[index] = arr[index], arr[i]


if __name__ == "__main__":
    arr = [1, 2, 4, 2, 1]
    selection_sort(arr)
    print(arr)
